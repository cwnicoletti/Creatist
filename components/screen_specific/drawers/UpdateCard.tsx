import React from "react";
import {
  Platform,
  StyleSheet,
  Text,
  TouchableNativeFeedback,
  TouchableOpacity,
  View,
} from "react-native";

const UpdateCard = (props) => {
  let Component = null;
  switch (props.iconFamily) {
    case "Entypo":
      const { Entypo } = require("@expo/vector-icons");
      Component = Entypo;
      break;
    case "EvilIcons":
      const { EvilIcons } = require("@expo/vector-icons");
      Component = EvilIcons;
      break;
    case "FontAwesome":
      const { FontAwesome } = require("@expo/vector-icons");
      Component = FontAwesome;
      break;
    case "MaterialIcons":
      const { MaterialIcons } = require("@expo/vector-icons");
      Component = MaterialIcons;
      break;
    case "Foundation":
      const { Foundation } = require("@expo/vector-icons");
      Component = Foundation;
      break;
    default:
      break;
  }

  let TouchableCmp: any = TouchableOpacity;
  if (Platform.OS === "android") {
    TouchableCmp = TouchableNativeFeedback;
  }

  return (
    <View style={{ ...styles.exhibit, ...props.exhibitContainer }}>
      {!props.iconFamily || !props.iconName || Component == null ? null : (
        <Component
          name={props.iconName}
          size={24}
          color={props.darkModeValue ? "white" : "black"}
        />
      )}
      <View style={styles.touchable}>
        <Text style={{ ...styles.title, ...props.titleStyle }}>
          {props.updateTitle}
        </Text>
        <Text style={{ ...styles.body, ...props.bodyStyle }}>
          {props.updateBody}
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  exhibit: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    alignSelf: "center",
    height: 100,
    width: "80%",
  },

  title: {
    fontWeight: "500",
    fontSize: 12,
    margin: 20,
    marginBottom: 10,
  },

  body: {
    fontWeight: "300",
    fontSize: 12,
    marginHorizontal: 20,
  },
  
  touchable: {
    overflow: "hidden",
  },
});

export default UpdateCard;
