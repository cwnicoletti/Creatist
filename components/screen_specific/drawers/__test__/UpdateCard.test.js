import React from "react";
import renderer from "react-test-renderer";
import UpdateCard from "../UpdateCard";

test("matches previous snapshot and renders correctly", () => {
  const tree = renderer.create(<UpdateCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
