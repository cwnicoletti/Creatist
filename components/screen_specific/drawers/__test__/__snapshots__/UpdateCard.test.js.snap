// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches previous snapshot and renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "center",
      "width": "80%",
    }
  }
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
          "fontWeight": "500",
          "margin": 20,
          "marginBottom": 10,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "fontWeight": "300",
          "marginHorizontal": 20,
        }
      }
    />
  </View>
</View>
`;
